samples/kubernetes/end2end/demo-domains/domainBuilder/build.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+# Copyright 2019, Oracle Corporation and/or its affiliates.  All rights reserved.
+# Licensed under the Universal Permissive License v 1.0 as shown at http://oss.oracle.com/licenses/upl.
+set -e 
+
+MYDIR="$(dirname "$(readlink -f "$0")")" # get the absolute path of this file's folder
+PRJ_ROOT=$MYDIR/../../../../..
+
+# To build the archive which contains binaries and scripts etc.
+# Refer to WDT doc about the archive format: https://github.com/oracle/weblogic-deploy-tooling#simple-example
+function createArchive() {
+  mkdir -p ${MYDIR}/archive/wlsdeploy/applications
+  mkdir -p ${MYDIR}/archive/wlsdeploy/domainLibraries
+
+  echo 'Build the test webapp...'
+  cd test-webapp && mvn clean install && cd ..
+  cp test-webapp/target/testwebapp.war ${MYDIR}/archive/wlsdeploy/applications/testwebapp.war
+
+  echo 'Build the metrics exporter...'
+  cd $PRJ_ROOT
+  mvn clean install
+  cd webapp
+  mvn clean package -Dconfiguration=$MYDIR/../../dashboard/exporter-config.yaml
+  cd $MYDIR 
+  cp $PRJ_ROOT/webapp/target/wls-exporter.war \
+     ${MYDIR}/archive/wlsdeploy/applications/wls-exporter.war
+
+  echo 'Build the WDT archive...'
+  jar cvf ${MYDIR}/archive.zip  -C ${MYDIR}/archive wlsdeploy
+}
+
+function cleanTmpDir() {
+  rm -rf ${MYDIR}/archive
+  rm -rf ${MYDIR}/test-webapp/target
+  rm -f  ${MYDIR}/archive.zip
+  rm -rf $PRJ_ROOT/webapp/target
+  rm -f $MYDIR/weblogic-deploy.zip
+}
+
+function buildImage() {
+  if [ ! -e $MYDIR/weblogic-deploy.zip ] ; then
+    echo "weblogic-deploy.zip does not exist. Downloading it from github."  
+    wget -P $MYDIR https://github.com/oracle/weblogic-deploy-tooling/releases/download/weblogic-deploy-tooling-0.11/weblogic-deploy.zip
```

Comment:
(1) 0.11 is an old version of the tool.
(2) The get can go wrong - it may need special proxy settings, etc.
(3) We should maintain the WDT download in a single location in the tree, so future changes/tweaks to the WDT download are easily made in a single location.
(4) Similarly we may want to do the same for the unzipping/invoking of the tool.
(5) See ./kubernetes/samples/scripts/create-weblogic-domain/domain-home-on-pv/wdt/create-domain-script.sh

Reply:
(1) 0.11 is an old version of the tool.
> Changed to use v0.24.

(2) The get can go wrong - it may need special proxy settings, etc.
> I don't want to handle this in the scripts. I add proxy setting in the guide: https://github.com/oracle/weblogic-monitoring-exporter/blob/revert-31-revert-30-end2end-sample/samples/kubernetes/end2end/docs/04-wls-domain.md#set-proxy-before-run

(3) We should maintain the WDT download in a single location in the tree, so future changes/tweaks to the WDT download are easily made in a single location.
(4) Similarly we may want to do the same for the unzipping/invoking of the tool.
(5) See ./kubernetes/samples/scripts/create-weblogic-domain/domain-home-on-pv/wdt/create-domain-script.sh
> I did some clean up to the scripts. Pls review it in new PR https://github.com/oracle/weblogic-monitoring-exporter/pull/32.